flask_caching/__init__.py

```diff
@@ -215,28 +216,28 @@ def init_app(self, app: Flask, config=None) -> None:
     def _set_cache(self, app: Flask, config) -> None:
         import_me = config["CACHE_TYPE"]
         if "." not in import_me:
-            from . import backends
-
-            try:
-                cache_obj = getattr(backends, import_me)
-            except AttributeError:
-                raise ImportError(
-                    "%s is not a valid Flask-Caching backend" % (import_me)
-                )
-        else:
-            cache_obj = import_string(import_me)
+            warnings.warn(
+                "Using the initialization functions in flask_caching.backend "
+                "is deprecated.  Use the a full path to backend classes "
+                "directly.",
+                category=DeprecationWarning)
+            import_me = type(self).__module__ + '.backends.' + import_me
```

Comment:
`type(self).__module__` is breaking any application that subclasses this class since it will then resolve to the module where the subclass is. So I think this should remain as a hardcoded `'flask_caching.backends.' + import_me`

Reply:
This line only runs if the cache backend you specify in your config (in `CACHE_TYPE`) doesn’t contain a dot (see line 218). If you define your own backend, it will be of the form `your_package.your_module.YourClass`, so `import_me` won’t be mangled for you.

Reply:
i was more thinking about the built-in cache types. `CACHE_TYPE = 'null'` for example. Why would I ever want that to resolve to something different than the built-in one?

Reply:
In that case, `null` gets expanded to `flask_caching.backends.null`; `self.module` is always `flask_caching` in this context; itʼs written this way so even if we rename the whole module, we donʼt have to modify it here.

Reply:
It isn't if you subclass the `Caching` class in your own module. Then it resolves to that one :/

Reply:
Ah, so thatʼs what you meant! Sorry for not getting it earlier.

Yes, you are right. I didnʼt consider that when writing the code because i couldnʼt imagine a use case for that. Unless you want to submit a fix, iʼll look into it later this week.